core/src/main/java/de/danoeh/antennapod/core/syndication/handler/FeedHandler.java

```diff
@@ -7,30 +7,61 @@
 import java.io.File;
 import java.io.IOException;
 import java.io.Reader;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
 
 import javax.xml.parsers.ParserConfigurationException;
 import javax.xml.parsers.SAXParser;
 import javax.xml.parsers.SAXParserFactory;
 
-import de.danoeh.antennapod.model.feed.Feed;
+import de.danoeh.antennapod.core.feed.Feed;
+import de.danoeh.antennapod.core.feed.FeedItem;
 
 public class FeedHandler {
 
-	public FeedHandlerResult parseFeed(Feed feed) throws SAXException, IOException,
-			ParserConfigurationException, UnsupportedFeedtypeException {
-		TypeGetter tg = new TypeGetter();
-		TypeGetter.Type type = tg.getType(feed);
-		SyndHandler handler = new SyndHandler(feed, type);
-
-		SAXParserFactory factory = SAXParserFactory.newInstance();
-		factory.setNamespaceAware(true);
-		SAXParser saxParser = factory.newSAXParser();
-		File file = new File(feed.getFile_url());
-		Reader inputStreamReader = new XmlStreamReader(file);
-		InputSource inputSource = new InputSource(inputStreamReader);
-
-		saxParser.parse(inputSource, handler);
-		inputStreamReader.close();
-		return new FeedHandlerResult(handler.state.feed, handler.state.alternateUrls);
-	}
+    public FeedHandlerResult parseFeed(Feed feed) throws SAXException, IOException,
+            ParserConfigurationException, UnsupportedFeedtypeException {
+        TypeGetter tg = new TypeGetter();
+        TypeGetter.Type type = tg.getType(feed);
+        SyndHandler handler = new SyndHandler(feed, type);
+
+        SAXParserFactory factory = SAXParserFactory.newInstance();
+        factory.setNamespaceAware(true);
+        SAXParser saxParser = factory.newSAXParser();
+        File file = new File(feed.getFile_url());
+        Reader inputStreamReader = new XmlStreamReader(file);
+        InputSource inputSource = new InputSource(inputStreamReader);
+
+        saxParser.parse(inputSource, handler);
+        inputStreamReader.close();
+        feed.setItems(dedupItems(feed.getItems()));
```

Comment:
Indent change.  This line is the only change. `feed.setItems(dedupItems(feed.getItems()));`